src/controllers/developers/Applications.ts

```diff
@@ -0,0 +1,156 @@
+import Responses from "../../utils/Responses";
+import SqlDatabase from "../../database/SqlDatabase";
+
+const crypto = require('crypto');
+
+/*
+ToDo
+Implement application limit
+Implement the token in oauth2
+ */
+
+export default class Applications {
+    public static applicationsLimit = 10;
+
+    public static async GET(req: any, res: any): Promise<void> {
+        const fetchApplications = await SqlDatabase.Execute("SELECT * FROM applications WHERE user_id = ? AND active = 1 ORDER BY id DESC", [req.user.id]);
+
+        Responses.Send(req, res, "developers/applications", "Applications | Quaver", {
+            applications: fetchApplications
+        });
+    }
+
+    public static async CreateGET(req: any, res: any): Promise<void> {
+        Responses.Send(req, res, "developers/application", "Create Application | Quaver", {
+            edit: false
+        });
+    }
+
+    public static async ViewGET(req: any, res: any): Promise<void> {
+        const applicationId = req.params.id;
+        const application = await Applications.ApplicationCheck(req, applicationId);
+
+        if(!application) {
+            req.flash('error', 'You are not the owner of this application!');
+            return res.redirect(303, `/developers/applications`);
+        }
+
+        const fetchApplication = await SqlDatabase.Execute("SELECT * FROM applications WHERE id = ? LIMIT 1", [applicationId]);
+
+        Responses.Send(req, res, "developers/application", "Application | Quaver", {
+            edit: true,
+            application: fetchApplication[0]
+        });
+    }
+
+    public static async SavePOST(req: any, res: any): Promise<void> {
+        const applicationId = req.body.application_id;
+        const name = await Applications.htmlEntities(req.body.application_name);
+        const redirectUrl = req.body.application_redirect_url;
+        let isEdit = false;
+
+        if(name === "" || redirectUrl === "") {
+            req.flash('error', 'Empty fields!');
+            // ToDo return form
+            return res.redirect(303, `/developers/applications/create`);
+        }
+
+        if (applicationId !== undefined) {
+            const application = await Applications.ApplicationCheck(req, applicationId);
+            if (!application) {
+                req.flash('error', 'You are not the owner of this application!');
```

Comment:
Perhaps move this check above, so the "Empty fields" error can also redirect back to editing if it's editing.